File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/TestCaseEpilogue.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class TestCaseEpilogue
    {
        public List<string> SubscribedTopics { get; set; } = new();

        public int? PublicationCount { get; set; }

        public List<TestCasePublishedMessage> PublishedMessages { get; set; } = new();

        public int? AcknowledgementCount {  get; set; }

        public List<TestCaseReceivedTelemetry> ReceivedTelemetries { get; set; } = new();

        public int? ExecutionCount { get; set; }

        public Dictionary<int, int> ExecutionCounts { get; set; } = new();

        public int? TelemetryCount { get; set; }

        public Dictionary<int, int> TelemetryCounts { get; set; } = new();

        public TestCaseCatch? Catch { get; set; }
    }
}

File: go/test/protocol/test_case_epilogue.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

type TestCaseEpilogue struct {
	SubscribedTopics     []string                    `yaml:"subscribed-topics"`
	PublicationCount     *int                        `yaml:"publication-count"`
	PublishedMessages    []TestCasePublishedMessage  `yaml:"published-messages"`
	AcknowledgementCount *int                        `yaml:"acknowledgement-count"`
	ReceivedTelemetries  []TestCaseReceivedTelemetry `yaml:"received-telemetries"`
	ExecutionCount       *int                        `yaml:"execution-count"`
	ExecutionCounts      map[int]int                 `yaml:"execution-counts"`
	TelemetryCount       *int                        `yaml:"telemetry-count"`
	TelemetryCounts      map[int]int                 `yaml:"telemetry-counts"`
	Catch                *TestCaseCatch              `yaml:"catch"`
}
